brozzler/worker.py

```diff
@@ -316,8 +316,10 @@ def _on_service_worker_version_updated(chrome_msg):
             page.note_redirect(final_page_url)
         return outlinks
 
-    def _fetch_url(self, site, url):
+    def _fetch_url(self, site, url=None, page=None):
```

Comment:
This `page` param could also by type-annotated (as could any other places you've added a `page` param)